ad_source/view_sets.py

```diff
@@ -173,3 +199,39 @@ def get(self, request):
             return Response(status=status.HTTP_403_FORBIDDEN)
         logout(request)
         return Response(status=status.HTTP_200_OK)
+
+
+class RewardViewSet(viewsets.ModelViewSet):
+    authentication_classes = [authentication.CsrfExemptSessionAuthentication, BasicAuthentication]
+    queryset = models.Reward.objects.all()
+
+    def create(self, request, *args, **kwargs):
+        task_id = kwargs['task_id']
+        task = get_object_or_404(models.Task, pk=task_id)
+        reward, created = models.Reward.objects.get_or_create(
+            receiver=request.user,
+            task=task,
+            defaults={
+                'sender': task.user,
+                'amount': task.reward_per_click,
+            }
+        )
+        checksummed_sender = Web3.toChecksumAddress(str(reward.sender.username))
+        checksummed_receiver = Web3.toChecksumAddress(str(reward.receiver.username))
+        if created:
+            transaction = contract_instance.functions.forwardRewards(
+                checksummed_receiver,  # To
+                checksummed_sender,  # From
+                task.website_link  # task's website url
+            ).buildTransaction({
+                'chainId': 80001,
+                'gas': 100000,
+                'gasPrice': w3.toWei('1', 'gwei'),
+                'nonce': w3.eth.getTransactionCount(settings.ACCOUNT_OWNER_PUBLIC_KEY)
+            })
+            txn_signed = w3.eth.account.signTransaction(transaction, private_key=settings.ACCOUNT_OWNER_PRIVATE_KEY)
+            tx_hash_hex = w3.eth.sendRawTransaction(txn_signed.rawTransaction)  # tx_hash
+            tx_hash = tx_hash_hex.hex()
+            return Response(data=tx_hash, status=status.HTTP_201_CREATED)
+        else:
+            return HttpResponseBadRequest(content='Incorrect web3 parameters')
```

Comment:
..and this response to:
```
data = {
"error": "Reward already created"
}
return HttpResponseBadRequest(json.dumps(data), content_type='application/json')
```

Reply:
..similar to my previous comment. Response should have data that are in JSON format.

```
data = {
    "error": "Reward already created"
}
return Response(data=data, status=status.HTTP_400_BAD_REQUEST)
```